crates/core/component/auction/src/component/trigger_data.rs

```diff
@@ -0,0 +1,156 @@
+use anyhow::Result;
+
+// A device that captures immutable trigger data description, and provides
+// safe helpers to compute the next trigger height, or the current step index.
+pub struct TriggerData {
+    pub start_height: u64,
+    pub end_height: u64,
+    pub step_count: u64,
+}
+
+impl TriggerData {
+    /// Compute the step index of the supplied height.
+    ///
+    /// # Errors
+    /// This method errors if the block interval is not a multiple of the
+    /// specified `step_count`, or if it operates over an invalid block
```

Comment:
The method doesn't actually error in this case though right? The `step_size` will be rounded down. I'm not sure it matters here but it doesn't quite match the description.